File: MyList.TestHelpers/ShoppingListItemBuilder.cs
using MyList.Entity;
using System;
using System.Collections.Generic;
using System.Text;

namespace MyList.TestHelpers
{
    public class ShoppingListItemBuilder
    {
        private Guid _id = Guid.NewGuid();
        private Guid _shoppingListId = Guid.NewGuid();
        private string _name = "DefaultItemName";
        private bool _isChecked = false;
        private int _sortOrder = 1;

        public ShoppingListItemBuilder()
        { }

        public static ShoppingListItemBuilder Create()
        {
            return new ShoppingListItemBuilder();
        }

        public ShoppingListItemBuilder WithId(Guid id)
        {
            _id = id;
            return this;
        }

        public ShoppingListItemBuilder WithIdShoppingListID(Guid id)
        {
            _shoppingListId = id;
            return this;
        }

        public ShoppingListItemBuilder WithName(string name)
        {
            _name = name;
            return this;
        }

        public ShoppingListItemBuilder WithIsChecked()
        {
            _isChecked = true;
            return this;
        }

        public ShoppingListItemBuilder WithSortOrder(int sortOrder)
        {
            _sortOrder = sortOrder;
            return this;
        }

        public ShoppingListItem Build()
        {
            return new ShoppingListItem() 
            { 
                Id = _id,
                ShoppingListId = _shoppingListId,
                Name = _name,
                IsChecked = _isChecked,
                SortOrder = _sortOrder
            };
        }
    }
}

File: MyList.Web/src/app/test/builders/shopping-list-item.builder.ts
import { ShoppingListItem } from "src/app/models/shopping-list-item";

export class ShoppingListItemBuilder {
  private _id: string = "1";
  private _name: string = "defaultName";
  private _shoppingListId: string = "1";
  private _isChecked: boolean = false;

  constructor() {}

  static create = () => new ShoppingListItemBuilder();

  withId(id: string): ShoppingListItemBuilder {
    this._id = id;
    return this;
  }

  withName(name: string): ShoppingListItemBuilder {
    this._name = name;
    return this;
  }

  withShoppingListId(id: string): ShoppingListItemBuilder {
    this._shoppingListId = id;
    return this;
  }

  withIsChecked(): ShoppingListItemBuilder {
    this._isChecked = true;
    return this;
  }

  build(): ShoppingListItem {
    return {
      id: this._id,
      name: this._name,
      shoppingListId: this._shoppingListId,
      isChecked: this._isChecked
    } as ShoppingListItem
  }
}
